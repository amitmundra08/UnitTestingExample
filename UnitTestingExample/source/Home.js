import React, {Component} from 'react';
import {View, Text, StyleSheet, TextInput} from 'react-native';
import Profile from './Profile';

class Home extends Component {
  constructor() {
    super();
    this.state = {
      data: 'String',
    };
  }

  onChangeValue = (x) => {
    this.setState({data: x * 10});
  };

  render() {
    return (
      <View style={styles.container}>
        <Profile data="hello world" />
        <TextInput
          testID={'username'}
          style={{height: 40, backgroundColor: 'grey'}}
          placeHolder="Enter Username"
          onChangeText={(value) => this.onChangeValue(value)}
        />
        <TextInput
          testID={'password'}
          style={{height: 40, backgroundColor: 'grey'}}
          placeHolder="Enter Password"
          onChangeText={(value) => this.onChangeValue(value)}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default Home;
