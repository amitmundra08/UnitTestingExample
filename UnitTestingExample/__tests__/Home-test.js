import 'react-native';
import React from 'react';
import Home from './../source/Home';
import renderer from 'react-test-renderer';

test('Home', () => {
  const snap = renderer.create(<Home />).toJSON();
  expect(snap).toMatchSnapshot();
});
