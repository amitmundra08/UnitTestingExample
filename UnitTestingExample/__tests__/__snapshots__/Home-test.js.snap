// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={Object {}}
  >
    <Text
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      hello world
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeHolder="Enter Username"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "grey",
        "height": 40,
      }
    }
    testID="username"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeHolder="Enter Password"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "grey",
        "height": 40,
      }
    }
    testID="password"
    underlineColorAndroid="transparent"
  />
</View>
`;
